src/Compilers/CSharp/Portable/Symbols/ConstraintsHelper.cs

```diff
@@ -438,6 +488,8 @@ private static bool CheckConstraintsSingleType(TypeSymbol type, CheckConstraints
             Location location,
             DiagnosticBag diagnostics)
         {
+            type = (NamedTypeSymbol)type.TupleUnderlyingTypeOrSelf();
```

Comment:
> I iterate over all elements and so on.

The confusing part is that you are not doing that here, in this function. You are doing that in in the previous function. In this function you simply "unwrap" a tuple. Why is this sufficient requires an explanation. The explanation I came up with after examining the code is that TypeSymbolExtensions.VisitType does the trick. If this explanation is not correct, please explain when and where do we check constraints for the "Rest" tuple, in the scenario, for which we hit this code path with a "long" tuple.